### DISCUSSION TITLE
关于必应的常见问题解答

### DISCUSSION THREAD

**Discussion by Redbeanw44602** [PenUniverse/PenMods#213] [发布专区]
先说结论：截止2024年3月21日下午1点10分，词典笔版必应**可用**，所有不可用情况均为您自身（代理）问题。

1. **如何搭建代理？**
请使用谷歌搜索关键词：`nginx 反向代理` `nginx 反向代理 websocket`

2. **要求您切换到 cn.bing.com**
必应认为你身处大陆，大陆的国内版必应没有AI对话支持。即使代理的IP确实不是国内的也可能出现，这可能是因为IP被标记导致的，建议更换服务商。

3. **网络异常，服务器未返回任何数据**
代理服务器触发必应风控，建议更换服务商。

4. **Json异常，显示大段HTML文本**
与2相同。

5. **Json异常，显示一段含conversationXX的小段Json格式文本**
如果你是1.3.0及以上，请不要在代理的请求参数中附加低版本bundleVersion参数，1.2.0及以下需要附加 `bundleVersion=1.0` 参数，建议更新PenMods。

6. **未验证的请求**
已知问题，重新发起提问即可，最近更新可能解决。

7. **开始回答前套接字错误N**
N为错误码，参考：https://doc.qt.io/qt-6/qabstractsocket.html#SocketError-enum
一般来讲出现这个问题大多是因为链接大小写未正确区分、协议（ws/wss）未写对，末位斜杠未正确加（或不加）导致的。
最近更新会提供更友好的错误提示。

8. **回答时套接字错误**
网络抖动，或文本太长渲染使用太多的资源，最近更新可能优化该问题。

其他问题请在下方留言，不定期更新。

**Comment by Tone-cloud** [comment]
怎么搭建

**Reply by GuangSeven** [reply]
STFW，看不懂人话？
